react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+
+const images = [
+  {
+    number: '001',
+    name: 'Bulbasaur',
+    src: '../images/001.png'
+  },
+  {
+    number: '004',
+    name: 'Charmander',
+    src: '../images/004.png'
+  },
+  {
+    number: '007',
+    name: 'Squirtle',
+    src: '../images/007.png'
+  },
+  {
+    number: '025',
+    name: 'Pikachu',
+    src: '../images/025.png'
+  },
+  {
+    number: '039',
+    name: 'Jigglypuff',
+    src: '../images/039.png'
+  }
+];
+
+function PokeImage(props) {
+  if (props.el === 'img') {
+    return (
+      <img src={props.src} alt={props.name} className={props.index === props.current ? '' : 'hidden'} />
+    );
+  } else {
+    return (
+      <i data-index={props.index} className={props.index === props.current ? 'fas fa-circle dot-column' : 'far fa-circle dot-column'}></i>
+    );
+  }
+}
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.changeInterval = this.changeInterval.bind(this);
+    this.dotMap = this.dotMap.bind(this);
+    this.currentImg = this.currentImg.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      index: 0
+    };
+    this.intervalId = setInterval(this.changeInterval, 3000);
```

Comment:
Use `componentDidMount` to start stuff like starting timers/data requests. The constructor runs first but if there's an error mounting your component, you don't want to cause a memory leak.